pylint/checkers/imports.py

```diff
@@ -792,28 +800,31 @@ def _report_dependencies_graph(self, sect, _, _dummy):
             _make_graph(filename, self._internal_dependencies_info(),
                         sect, 'internal ')
 
+    def _filter_dependencies_graph(self, internal):
+        """build the internal or the external depedency graph"""
+        graph = collections.defaultdict(set)
+        for importee, importers in self.stats['dependencies'].items():
+            for importer in importers:
+                package = self._module_pkg.get(importer, importer)
+                is_inside = importee.startswith(package)
+                if is_inside and internal or not is_inside and not internal:
+                    graph[importee].add(importer)
+        return graph
+
     def _external_dependencies_info(self):
         """return cached external dependencies information or build and
         cache them
         """
         if self.__ext_dep_info is None:
-            package = self.linter.current_name
-            self.__ext_dep_info = result = {}
-            for importee, importers in self.stats['dependencies'].items():
-                if not importee.startswith(package):
-                    result[importee] = importers
+            self.__ext_dep_info = self._filter_dependencies_graph(False)
```

Comment:
Let's call this with `is_inside=False` so it's clear when reading the code what `False` stands for.

Reply:
The argument name is `internal`, did you mean `internal=False`?

Reply:
Yes, `internal`!